halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/Providers.java

```diff
@@ -46,16 +51,33 @@
   GoogleProvider google = new GoogleProvider();
   KubernetesProvider kubernetes = new KubernetesProvider();
   OpenstackProvider openstack = new OpenstackProvider();
+  @JsonProperty(access = Access.WRITE_ONLY)
   OracleBMCSProvider oraclebmcs = new OracleBMCSProvider();
+  OracleProvider oracle = new OracleProvider();
 
   @Override
   public String getNodeName() {
     return "provider";
   }
 
+  public OracleProvider getOracle() {
+    return OracleProvider.mergeOracleBMCSProvider(oracle, oraclebmcs);
+  }
+  
   @Override
   public NodeIterator getChildren() {
-    return NodeIteratorFactory.makeReflectiveIterator(this);
+    List<Node> nodes = new ArrayList<Node>();
```

Comment:
Can we keep using reflective iterator, but maybe extend it to take preconditions? e.g. name != oraclebmcs. Otherwise this just adds to the overhead for adding a new provider.